scripts/sync-requirements-txt.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+set -Eeuxo pipefail
+
+# Our build tooling depends on requirements.txt files with hashes
+# Namely, Konflux (https://konflux-ci.dev/), and Cachi2 (https://github.com/containerbuildsystem/cachi2).
+
+# The following will create an extra requirement.txt file for every Pipfile.lock we have.
+micropipenv --version || pip install micropipenv
+find . -name Pipfile.lock -execdir bash -c '
+  echo "# Generated by /scripts/sync-requirements-txt.sh from Pipfile.lock" > requirements.txt &&
+  echo >> requirements.txt &&
```

Comment:
There is also this command `pipenv requirements --hash > requirements.txt` that can generate the requirements.txt with hashes. Did you try it?

Reply:
Yes, I started with that, but changed later.

In our Dockerfiles we use micropipenv, so I think its a better choice. Also I really is micro, so it is smaller and faster. And in the requirements.txt it generates, it formats them better, with indentation. And adds hashes by default, pipenv requirements needs extra flag.

Reply:
When we're building Dockerfiles now, it prints things like

```
Collecting torch==2.4.1+rocm6.1 (from -r /tmp/requirements_micropipenv-w76_pwe3.txt (line 2))
  Downloading https://download.pytorch.org/whl/rocm6.1/torch-2.4.1%2Brocm6.1-cp311-cp311-linux_x86_64.whl (2662.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 2.7/2.7 GB 272.0 MB/s eta 0:00:00
```

because micropipenv takes Pipfile.lock and first generates `requirements.txt` into a `/tmp` file and then runs pip to install from it.